src/app/core/services/issue.service.ts

```diff
@@ -329,22 +331,71 @@ export class IssueService {
     // const issuesAPICallsByFilter = filters.map(filter => this.githubService.fetchIssuesGraphql(filter));
     return forkJoin(issuesAPICallsByFilter).pipe(
       map((issuesByFilter: [][]) => {
+        const fetchedIssueIds: Array<Number> = [];
+
         for (const issues of issuesByFilter) {
           for (const issue of issues) {
+            fetchedIssueIds.push(this.createIssueModel(issue).id);
             this.createAndSaveIssueModel(issue);
           }
         }
+
+        return fetchedIssueIds;
+      }),
+      reduce((acc, ids) => acc.concat(ids), []),
+      map((ids: Array<Number>) => {
+        const outdatedIssueIds: Array<Number> = this.getOutdatedIssueIds(ids);
+        this.deleteIssuesFromLocalStore(outdatedIssueIds);
+
         return Object.values(this.issues);
       })
     );
   }
 
+
   private createAndSaveIssueModel(githubIssue: GithubIssue): boolean {
     const issue = this.createIssueModel(githubIssue);
     this.updateLocalStore(issue);
     return true;
   }
 
+  private deleteIssuesFromLocalStore(ids: Array<Number>): void {
+    ids.forEach((id: number) => {
+      this.getIssue(id).subscribe(issue => this.deleteFromLocalStore(issue));
+    });
+  }
+
+  /**
+   * Returns an array of outdated issue ids by comparing the ids of the recently
+   * fetched issues with the current issue ids in the local store
+   */
+  private getOutdatedIssueIds(fetchedIssueIds: Array<Number>): Array<Number> {
+    const result = [];
+
+    /*
+      Ignore for first fetch or ignore if there is no fetch result
+
+      We also have to ignore for no fetch result as the cache might return a
+      304 reponse with no differences in issues, resulting in the fetchIssueIds
+      to be empty
+    */
+    if (this.issues === undefined || !fetchedIssueIds.length) {
+      return result;
+    }
+
+    const fetchedIssueIdsSet = new Set<Number>(fetchedIssueIds);
+
+    const originalIssueIds = Object.keys(this.issues);
+
+    originalIssueIds.forEach((issueId: string) => {
+      if (!fetchedIssueIdsSet.has(+issueId)) {
+        result.push(+issueId);
+      }
```

Comment:
(This maybe very subjective)
TIL you can convert string to number in javascript by using `+` operator. 😄 
Perhaps, it's better to use `Number(issueId)` instead?

Reply:
I saw a few other places using the + sign to typecast string to Number, like inside issue.model.ts, hence I thought it would be better to adopt the same style. Not sure if it would be better to use Number for this. Any suggestions @CATcher-org/2021-devs ?

Reply:
I didn't notice we use the + sign in other places. TIL 😄 .
If that's the case then I think it's better to use the + sign here since it would make it consistent across the codebase.